### PR TITLE
Litera certified-connector files

### PR DESCRIPTION
**Pull request by momshead**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [X ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [X ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [X ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ X] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [X ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Review by vmanoharas — COMMENTED** [review]
Hello @momshead 

please make sure icon size should be 230x230 Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector).

File: certified-connectors/Litera/Readme.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Litera Copilot Plugin
```
Comment (vmanoharas) [inline certified-connectors/Litera/Readme.md]: Please make sure that the readme.md file format should be in the below format and all the required fields should be present.

File: certified-connectors/Litera/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,126 @@
+{
+	"swagger": "2.0",
+	"info": {
+		"title": "Litera.Lattice.Chat.Orchestrator.Api",
```
Comment (vmanoharas) [inline certified-connectors/Litera/apiDefinition.swagger.json]: Please make sure that the title character length should not cross 30 characters and should not contain reserved keyword API.  Please refer below link. 

https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#give-your-connector-a-title

File: certified-connectors/Litera/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+				"responses": {
+					"200": {
+						"description": "default",
+						"schema": {}
+					}
+				},
+				"produces": [
+					"application/json"
+				],
+				"summary": "Allows the user to search for legal matters (aka deals, transactions, or cases) and their associated IDs.  Returns a list of matter names and matter IDs that satisfy the specified user natural language request for matters.",
```
Comment (vmanoharas) [inline certified-connectors/Litera/apiDefinition.swagger.json]: Please reduce summary values below 80 characters and make sure that summary value ends with alphanumeric characters.

File: certified-connectors/Litera/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+					{
+						"name": "matterId",
+						"description": "The matter ID (matter number) of the matter for which to retrieve the narrative.",
+						"in": "query",
+						"type": "string",
+						"required": true,
+						"x-ms-summary": "The matter ID (matter number) of the matter for which to retrieve the narrative."
+					}
+				],
+				"summary": "Allows the user to retrieve a matter (aka deal, transaction, or case) narrative -- i.e. the descriptive overview -- for the specified matter.  Returns the narrative.",
```
Comment (vmanoharas) [inline certified-connectors/Litera/apiDefinition.swagger.json]: Please reduce summary values below 80 characters and make sure that summary value ends with alphanumeric characters.

File: certified-connectors/Litera/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+				"produces": [
+					"application/json"
+				],
+				"responses": {
+					"200": {
+						"description": "default",
+						"schema": {}
+					}
+				},
+				"summary": "Allows the user to retrieve details of a matter (aka deal, transaction, or case). Returns the matter detail for the specified legal matter ID.",
```
Comment (vmanoharas) [inline certified-connectors/Litera/apiDefinition.swagger.json]: Please reduce summary values below 80 characters and make sure that summary value ends with alphanumeric characters.

File: certified-connectors/Litera/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,126 @@
+{
```
Comment (vmanoharas) [inline certified-connectors/Litera/apiDefinition.swagger.json:1]: Contact property and x-ms-connector-metadata property are required, please see the template below. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json#L7

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json#L25

**Comment by momshead** [comment]
@microsoft-github-policy-service agree company="Litera"

**Review by vmanoharas — COMMENTED** [review]
Hello @modery, 

Kindly resolve review comments. 

Additionally, the swagger file provided in the ISV or master branch does not contain openAI annotations. To enable copilot openAI plugin, there needs to be specific annotations stating  at least for one API, the openAI data is enabled.
Example :-
   "x-ms-openai-data": {
          "openai-enabled": true,
          "operations": [
          {
            "operationId": "FindMultipleDocuments",
            "x-ms-require-user-confirmation": true,
         }
      ]
}
In addition to the above openAI annotations, M365 Copilot also good to have x-ms-keywords: [list_of_keywords], included in the info section of the swagger. This (in addition to the connector displayName) is currently used to choose which plugins to use as potential candidates from the list of plugins available and is typically required for a plugin to be successfully used in M365 bizchat. 
Example :- 
"info": {
    "version": "1.0",
    "title": "MongoDB",
    "description": " "MongoDB is a general purpose, distributed, document data platform which is loved by developers and used across enterprises to run variety of workloads.You can now explore the various endpoints and perform CRUD operations and aggregations on your collection using the MongoDB Connector.",
    "x-ms-api-annotation": {
      "status": "Production"
    },
    "x-ms-keywords": [
      "Data Source”, 
      "Database Name",
      "Cluster Name",
      "Collection Name",
      "Data Object,
      "Document Database",
    ],
},

Guidelines for adding keywords: ensure the name of the service is a keyword, add a space-separated one as well if needed. Add keywords for words that will likely get used in the prompt. Add keywords for operations as well and add 2 each: 1 singular and 1 plural. Ex: for a route to get tickets, add a “ticket” and “tickets” keyword. Note that each array value is being matched exactly. So, if you have "purchase order" as a value or if the prompt only contains purchase or only contains order, it won’t match. Matching is case-insensitive.
Preferred to add ‘x-ms-keywords’ to your swagger file.

**Comment by momshead** [comment]
@vmanoharas  - we are just finalizing our testing prompts, and will have them to you 

> Hello @momshead
> 
> please make sure icon size should be 230x230 Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector).

updating file now.

File: certified-connectors/Litera/Readme.md (hunk 1 above)
Comment (momshead) [reply certified-connectors/Litera/Readme.md]: questions for you:
1. what is "x-ms-require-user-confirmation", and should we just set it to true for each operation?
2. should we include the following:
"x-ms-api-annotation": {
      "status": "Production"
    },

File: certified-connectors/Litera/apiDefinition.swagger.json (hunk 2 above)
Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json]: OK

File: certified-connectors/Litera/apiDefinition.swagger.json (hunk 3 above)
Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json]: OK

File: certified-connectors/Litera/apiDefinition.swagger.json (hunk 6 above)
Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json:1]: i could use descriptions for each of the attributes for which we need values for x-ms-connector-metadata.

Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json:1]: please provide

Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json:1]: nevermind - i found it online

Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json:1]: where do I find a list of categories

File: certified-connectors/Litera/apiDefinition.swagger.json (hunk 5 above)
Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json]: done - will be reflected in updated submission.

File: certified-connectors/Litera/apiDefinition.swagger.json (hunk 2 above)
Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json]: done - will be reflected in updated submission.

File: certified-connectors/Litera/Readme.md (hunk 1 above)
Comment (momshead) [reply certified-connectors/Litera/Readme.md]: there is this question for you ^^.

File: certified-connectors/Litera/apiDefinition.swagger.json (hunk 6 above)
Comment (vmanoharas) [reply certified-connectors/Litera/apiDefinition.swagger.json:1]: This comment is for Contact and x-ms-connector-metadata both can find from the below template, can you please check this link. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json

File: certified-connectors/Litera/Readme.md (hunk 1 above)
Comment (vmanoharas) [reply certified-connectors/Litera/Readme.md]: No, please ignore this and add as below. 

"x-ms-api-annotation": {
      "status": "Preview"

Comment (momshead) [reply certified-connectors/Litera/Readme.md]: OK - done.  Will be reflected in next update.

File: certified-connectors/Litera/apiDefinition.swagger.json (hunk 6 above)
Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json:1]: done - reflected in updated fileset.

File: certified-connectors/Litera/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+				"responses": {
+					"200": {
+						"description": "default",
+						"schema": {}
+					}
+				},
+				"produces": [
+					"application/json"
+				],
+				"summary": "Allows the user to search for legal matters and their associated IDs.",
```
Comment (vmanoharas) [inline certified-connectors/Litera/apiDefinition.swagger.json]: 'summary' value must end with alphanumeric character. ValuePath: GetMatterList/get/summary

File: certified-connectors/Litera/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+					{
+						"name": "matterId",
+						"description": "The matter ID (matter number) of the matter for which to retrieve the narrative.",
+						"in": "query",
+						"type": "string",
+						"required": true,
+						"x-ms-summary": "The matter ID (matter number) of the matter for which to retrieve the narrative."
+					}
+				],
+				"summary": "Allows the user to retrieve a matter narrative for the specified legal matter.",
```
Comment (vmanoharas) [inline certified-connectors/Litera/apiDefinition.swagger.json]: 'summary' value must end with alphanumeric character. ValuePath: GetMatterNarrative/get/summary

File: certified-connectors/Litera/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+				"produces": [
+					"application/json"
+				],
+				"responses": {
+					"200": {
+						"description": "default",
+						"schema": {}
+					}
+				},
+				"summary": "Allows the user to retrieve details of a legal matter.",
```
Comment (vmanoharas) [inline certified-connectors/Litera/apiDefinition.swagger.json]: 'summary' value must end with alphanumeric character. ValuePath: GetMatterDetail/get/summary

File: certified-connectors/Litera/Readme.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+* Timekeeper
+* Responsible Attorney
+* Billing Attorney
+* Matter Office 
+* Matter Kind (open/closed) 
+* Total Hours (secured) 
+* Total Billing (secured) 
+* Adjudicator (judge name)
+* Forum
+
```
Comment (vmanoharas) [inline certified-connectors/Litera/Readme.md:43]: Kindly add obtaining credentials which is required field. 
https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md#obtaining-credentials

**Review by vmanoharas — COMMENTED** [review]
Hello @momshead, 

Can you please fix the comments?

File: certified-connectors/Litera/apiDefinition.swagger.json (hunk 7 above)
Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json]: done - will be reflected in next commit.

File: certified-connectors/Litera/apiDefinition.swagger.json (hunk 8 above)
Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json]: done - will be reflected in next commit.

File: certified-connectors/Litera/apiDefinition.swagger.json (hunk 9 above)
Comment (momshead) [reply certified-connectors/Litera/apiDefinition.swagger.json]: done - will be reflected in next commit.

File: certified-connectors/Litera/Readme.md (hunk 10 above)
Comment (momshead) [reply certified-connectors/Litera/Readme.md:43]: done - reflected in most recent commit.

**Comment by momshead** [comment]
All requests have been addressed, and a new commit has been pushed.  Please contact me asap for additional requests.  Thanks.

**Review by vmanoharas — APPROVED** [review]
Hello @momshead 
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to Connector certification portal](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#update-submission-prior-to-certification) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in [Connector certification portal](https://connectorcertification.azurewebsites.net/) . Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in [Connector certification portal](https://connectorcertification.azurewebsites.net/), please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector.
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.

**Pull request merged**